FEM-2594 netkit network calls retry

giladna opened the pull request.

giladna committed:
support in retry

giladna committed:
remove uneede import

giladna committed:
remove uneede import

giladna committed:
add default queue in order to keep old behaviour

giladna committed:
Merge branch 'develop' into FEM-2594

giladna committed:
add severity

giladna committed:
fix typo

giladna committed:
DateUtils.SECOND_IN_MILLIS instead of 1000

giladna committed:
expand RetryPolicy usage

giladna committed:
cosmetics

giladna committed:
add NetworkEventListener
fix typo

giladna committed:
add more error events

giladna committed:
add message to connect/call errors
cleanup

noamtamim committed:
Update RequestBuilder.java

noamtamim commented on netkit/src/main/java/com/kaltura/netkit/utils/NetworkEventListener.java:
```diff
@@ -0,0 +1,6 @@
+package com.kaltura.netkit.utils;
+
+public interface NetworkEventListener {
```
Rename to NetworkErrorListener.

giladna committed:
code review comments

giladna committed:
Code review changes
make RequestConfiguration a class
make rename NetworkEvent to NetworkErrorEvent
rename retryAttempts to maxRetries

giladna committed:
remove severity

giladna committed:
remove space

giladna committed:
uncomment super call
